### PR TITLE
Speeding up the pixel transform

### PR DESCRIPTION
**Pull request by ngreenwald**
The current implementation of the pixel transform first identifies all unique label IDs in all stacks of training images, then loops over each ID, then over each crop. 

The current implementation of the watershed transform instead first loops over each crop, then identifies the unique cell IDs within each crop, and then loops over each unique label ID.

The latter results in significant speed increases, as only the label IDs within each crop are iterated over. 

This PR refactors the pixel transform in the same style as watershed. 

Obviously a bug here would have far-reaching consequences, so please feel free to go over this with a fine-toothed comb. Happy to make any changes!

### CODE REVIEW

**Comment by willgraf** [comment]
I haven't had time to review the changes, but it looks like the refactor is not passing the `test_pixelwise_transform_3d` or the `test_pixelwise_transform_3d` tests in `deepcell/utils/transform_utils_test.py`.

You can test this locally using `pytest --cov=deepcell/utils --pep8`.  Make sure you install the `requirements-test.txt` dependencies first.

File: deepcell/utils/transform_utils.py [hunk 1]
```diff
@@ -55,86 +55,70 @@ def pixelwise_transform(mask, dilation_radius=None, data_format=None,
...
         all_stacks = [
-            edges,
-            interiors,
+            edge,
+            interior,
             background
         ]
 
-        return np.stack(all_stacks, axis=channel_axis)
+        return np.stack(all_stacks, axis=-1)
```
Comment (willgraf) [inline deepcell/utils/transform_utils.py]: We still pass in `data_format`, so why are we hardcoding `-1` here?

File: deepcell/utils/transform_utils.py [hunk 2]
```diff
@@ -55,86 +55,70 @@ def pixelwise_transform(mask, dilation_radius=None, data_format=None,
...
-        interior_edges,
-        interiors,
+        background_edge,
+        interior_edge,
+        interior,
         background
     ]
 
-    return np.stack(all_stacks, axis=channel_axis)
+    return np.stack(all_stacks, axis=-1)
```
Comment (willgraf) [inline deepcell/utils/transform_utils.py]: Again, why not leave it as `channel_axis`?

File: deepcell/image_generators/__init__.py [hunk 3]
```diff
@@ -92,10 +92,25 @@ def _transform_masks(y, transform, data_format=None, **kwargs):
...
+        for batch in range(y_transform.shape[0]):
+            if data_format == 'channels_first':
+                mask = y[batch, 0, ...]
+            else:
+                mask = y[batch, ..., 0]
+
+            y_transform[batch] = transform_utils.pixelwise_transform(mask, dilation_radius, data_format=data_format,
+                                                                     separate_edge_classes=separate_edge_classes)
+
+            if data_format == 'channels_first':
```
Comment (willgraf) [inline deepcell/image_generators/__init__.py]: This `rollaxis` is probably unnecessary if we generalize the transform with `data_format`.

File: deepcell/utils/transform_utils.py (hunk 2 above)
Comment (ngreenwald) [reply deepcell/utils/transform_utils.py]: I removed the channel_axis argument to mirror the logic of the watershed transform, where it's evaluated within the image_generators __init__.py but then not passed into the function: https://github.com/vanvalenlab/deepcell-tf/blob/b78caac6263d46ef1d1570110cb647df0a5a03ed/deepcell/image_generators/__init__.py#L115

If you'd rather not make that same change here happy to revert back

File: deepcell/utils/transform_utils_test.py [hunk 4]
```diff
@@ -105,34 +105,40 @@ def test_pixelwise_transform_3d(self):
             batch_count = maskstack.shape[0] // frames
```
Comment (ngreenwald) [inline deepcell/utils/transform_utils_test.py]: I wasn't sure why the pixel_transform test went with this approach, compared to the watershed_transform test, so I just looped through the stack for now

**Comment by ngreenwald** [comment]
The tests were failing because of the data format. I changed the tests to also loop through each test image, rather than passing a test image stack.

File: deepcell/utils/transform_utils_test.py (hunk 4 above)
Comment (willgraf) [reply deepcell/utils/transform_utils_test.py]: I also do not know why this approach was used, I think what you have done is fine.

**Comment by willgraf** [comment]
Hey this looks great! Thanks for making the changes.  I am going to update from master and merge it in when ready.

Great work!

**Pull request merged**